src/wp-includes/class-wp-scripts.php

```diff
@@ -794,18 +850,32 @@ public function in_default_dir( $src ) {
 	 * @return bool True on success, false on failure.
 	 */
 	public function add_data( $handle, $key, $value ) {
-		if ( 'strategy' === $key && ! empty( $value ) && ! $this->is_delayed_strategy( $value ) ) {
-			_doing_it_wrong(
-				__METHOD__,
-				sprintf(
+		if ( 'strategy' === $key ) {
+			if ( ! empty( $value ) && ! $this->is_delayed_strategy( $value ) ) {
+				_doing_it_wrong(
+					__METHOD__,
+					sprintf(
 					/* translators: 1: $strategy, 2: $handle */
-					__( 'Invalid strategy `%1$s` defined for `%2$s` during script registration.' ),
-					$value,
-					$handle
-				),
-				'6.3.0'
-			);
-			return false;
+						__( 'Invalid strategy `%1$s` defined for `%2$s` during script registration.' ),
+						$value,
+						$handle
+					),
+					'6.3.0'
+				);
+				return false;
+			} elseif ( empty( $this->registered[ $handle ]->src ) && $this->is_delayed_strategy( $value ) ) {
+				_doing_it_wrong(
+					__METHOD__,
+					sprintf(
+						/* translators: 1: $strategy, 2: $handle */
+						__( 'Cannot supply a strategy `%1$s` for script `%2$s` since it is a dependency bundle.' ),
```

Comment:
I had to read this a few times to understand the warning because I wasn't sure what a "dependency bundle" was referring to. Consider something like:

```suggestion
						__( 'Cannot supply a strategy `%1$s` for script `%2$s` because it does not have a `src` value.' ),
```